modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/type/IgniteTypeSystem.java

```diff
@@ -38,4 +40,36 @@
     @Override public int getMaxNumericPrecision() {
         return Short.MAX_VALUE;
     }
+
+    /** {@inheritDoc} */
+    @Override public RelDataType deriveSumType(RelDataTypeFactory typeFactory, RelDataType argumentType) {
+        RelDataType sumType;
+
+        switch (argumentType.getSqlTypeName()) {
+            case INTEGER:
+            case TINYINT:
+            case SMALLINT:
+                sumType = typeFactory.createJavaType(Long.class);
```

Comment:
Is there any reason to create java types but not SQL types? The overridden method creates SQL types.

Reply:
Fixed by add separate branch for the `BasicSqlType`